clang/include/clang/3C/ProgramInfo.h

```diff
@@ -158,9 +158,9 @@ class ProgramInfo : public ProgramVariableAdder {
                                                      ASTContext *C) const;
 
   void constrainWildIfMacro(ConstraintVariable *CV, SourceLocation Location,
-                            PersistentSourceLoc *PSL = nullptr);
+                            const ReasonLoc &Rsn);
```

Comment:
It looks like all calls to `constrainWildIfMacro` set the reason to `MACRO_REASON`.  Could `constrainWildIfMacro` take only a PSL (as before) and add `MACRO_REASON` internally?

Reply:
I'd prefer that we maintain `ReasonLoc`s for all PSLs (except where they are obviously incompatible), associating them with user feedback. This also gives us the option to have a more significant reason provided that `constrainWildIfMacro` would have no access to. This is one of the first PRs about user feedback, so let's not optimize too early.